packages/ngtools/webpack/src/gather_diagnostics.ts

```diff
@@ -9,6 +9,11 @@ import { Diagnostic, Diagnostics, Program } from '@angular/compiler-cli';
 import * as ts from 'typescript';
 import { time, timeEnd } from './benchmark';
 
+export const enum DiagnosticMode {
+  All,
```

Comment:
Is const needed? Also, you could use flags so it's possible to make combinations. Unless this is meant to be used as an internal only API, which it doens't look like.

Something like:

```
export enum DiagnosticMode {
  Syntactic = 1 << 0,
  Semantic = 1 << 1,

  All = Syntactic | Semantic,
  Default = All,
}
```

What do you think?

Reply:
I think that's much better really. Updated, thank you!